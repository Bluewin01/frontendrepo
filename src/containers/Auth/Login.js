import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import { Form, Button, Input } from "antd";
import { throttle } from "lodash";
import "./_loginStyle.css";
import { createStructuredSelector } from "reselect";
import * as AuthActions from "../../store/Auth/AuthActions";
import * as LoaderActions from "../../store/Loader/LoaderAction";
import PropTypes from "prop-types";
import Loader from "../../components/Loader/Loader";
import {
  makeSelectOTPData,
  makeSelectError,
  makeSelectJwt,
} from "../../store/Auth/AuthReselect";
import History from "../../routes/History";
import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";

const Login = (props) => {
  const {
    generateToken,
    otpData,
    sendVerifyOTP,
    error,
    loading,
    showLoader,
    generateStaffToken,
    jwt,
  } = props;
  const [isOTP, setIsOTP] = useState(false);
  const [isResend, setIsResend] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");

  useEffect(() => {
    if (jwt) {
      showLoader(false);
      History.push("/home");
    }
  }, [jwt, showLoader]);

  useEffect(() => {
    if (error !== null) {
      setErrorMessage(error.message);
    }
  }, [error]);

  useEffect(() => {
    if (otpData !== null) {
      setIsOTP(true);
    }
  }, [otpData]);

  function validateUserInput(userID, password) {
    let regex = new RegExp("[^a-zA-Z0-9]");

    if (
      (userID,
      password === null || userID,
      password === "" || userID,
      password === undefined || password === "")
    ) {
      setErrorMessage("Please fill in your ID or password");
    } else {
      if (userID.match(regex)) {
        setErrorMessage("Invalid ID");
      } else {
        verifyUserType(userID, password);
      }
    }
  }

  function verifyUserType(userID, password) {
    if (userID.charAt(0) === "I") {
      generateStaffToken(userID, password);
    } else {
      if (userID.length === 5) {
        generateToken(userID, password);
        showLoader(true);
        setErrorMessage("");
      } else {
        setErrorMessage("Invalid ID");
      }
    }
  }

  const handleOnSubmit = (data) => {
    const { userID, password } = data;
    validateUserInput(userID, password);
  };

  const handleOnOTPSubmit = (data) => {
    const { OTPCode } = data;
    sendVerifyOTP(otpData.id, OTPCode);
  };

  const throtleOnSubmitHandler = throttle(handleOnSubmit, 1000 * 30);

  return (
    <div className="login">
      {!isOTP ? (
        <>
          <h3>Login to NTP </h3>
          <Form
            className="login__form"
            onFinish={throtleOnSubmitHandler}
            layout="vertical"
          >
            <div className="input__form">
              <Form.Item
                style={{
                  fontSize: "20px",
                }}
                label="User ID"
                name="userID"
              >
                <Input
                  data-testid="userID"
                  placeholder="User ID"
                  autoComplete="off"
                ></Input>
              </Form.Item>
            </div>
            <div className="input__form">
              <Form.Item label="Password" name="password">
                <Input
                  data-testid="password"
                  type="password"
                  placeholder="Password"
                  width="100%"
                  autoComplete="off"
                ></Input>
              </Form.Item>
            </div>
            <div className="input__form">
              <Form.Item>
                <Button
                  data-testid="loginButton"
                  htmlType="submit"
                  danger
                  className="btn__primary"
                  size="large"
                >
                  Login
                </Button>
              </Form.Item>
            </div>
          </Form>
        </>
      ) : (
        <>
          <h3>Login to NTP</h3>
          <div style={{ fontSize: "16px", margin: "16px 0" }}>
            OTP code has been sent to your mobile number
          </div>
          <Form
            className="login__form"
            onFinish={handleOnOTPSubmit}
            layout="vertical"
          >
            <div className="input__form">
              <Form.Item label="OTP Code" name="OTPCode">
                <Input
                  data-testid="otp"
                  placeholder="OTP Code"
                  autoComplete="off"
                ></Input>
              </Form.Item>
            </div>
            <div className="input__form">
              <Form.Item>
                <Button
                  data-testid="otpButton"
                  htmlType="submit"
                  danger
                  className="btn__primary"
                  size="large"
                >
                  {isResend ? `Resend OTP` : "Verify OTP"}
                </Button>
              </Form.Item>
            </div>
          </Form>
        </>
      )}
      {loading ? <Loader></Loader> : null}
      {<ErrorMessage error={errorMessage} />}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  otpData: makeSelectOTPData(),
  error: makeSelectError(),
  jwt: makeSelectJwt(),
});

const mapDispatchToProps = (dispatch) => {
  return {
    generateToken: (userID, password) =>
      dispatch(AuthActions.generateTokenRequest({ userID, password })),
    showLoader: (isLoading) => dispatch(LoaderActions.loader(isLoading)),
    sendVerifyOTP: (id, code) =>
      dispatch(AuthActions.verifyOTPRequest({ id, code })),
    generateStaffToken: (userID, password) =>
      dispatch(AuthActions.generateStaffTokenRequest({ userID, password })),
  };
};

Login.protoTypes = {
  generateToken: PropTypes.func,
  generateAuthToken: PropTypes.object,
  otpData: PropTypes.object,
  sendVerifyOTP: PropTypes.func,
  error: PropTypes.object,
  loading: PropTypes.bool,
  generateStaffToken: PropTypes.func,
  jwt: PropTypes.object,
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
