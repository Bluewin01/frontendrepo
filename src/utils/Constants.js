export const theme = {
  color: {
    w: "#FFFFFF",
    k: "#000000",
    p1: "#D31145",
    p2: "#596C80",
    p3: "#554344",
    p4: "#363E3F",
    p5: "#e6edeb",
    s1: "#E86487",
    s2: "#FBBF56",
    s3: "#97CB5D",
    s4: "#46D3BD",
    s5: "#22A8DA",
    s6: "#9962D2",
    t1: "#F6CCD7",
    t2: "#B8123E",
    t3: "#F9E1B8",
    t4: "#C36418",
    t5: "#C6EB9E",
    t6: "#65962E",
    b1: "#D4EDF1",
    b2: "#FAF9F3",
    b3: "#F0EEE1",
    b4: "#DBDAD4",
    b5: "#B3B1A8",
    b6: "#6F6E68",
    b8: "#302F2C",
    b9: "#292025",
    u2: "#FF5500",
    u3: "#405B70",
    u4: "#769FAB",
  },
};

export const BASE_API = `${process.env.REACT_APP_BASE_API_URL}`;
