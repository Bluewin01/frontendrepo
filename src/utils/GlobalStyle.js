import { createGlobalStyle } from 'styled-components';
import 'antd/dist/antd.css';
import '@aia-digital/ui-library/lib/static/styles.css';

import AIABoldEOT from '../assets/fonts/AIAEverest-Bold.eot';
import AIABoldWOFF from '../assets/fonts/AIAEverest-Bold.woff';
import AIABoldWOFF2 from '../assets/fonts/AIAEverest-Bold.woff2';
import AIABoldTTF from '../assets/fonts/AIAEverest-Bold.ttf';
import AIABoldSVG from '../assets/fonts/AIAEverest-Bold.svg';

import AIACondensedEOT from '../assets/fonts/AIAEverest-Condensed.eot';
import AIACondensedWOFF from '../assets/fonts/AIAEverest-Condensed.woff';
import AIACondensedWOFF2 from '../assets/fonts/AIAEverest-Condensed.woff2';
import AIACondensedTTF from '../assets/fonts/AIAEverest-Condensed.ttf';
import AIACondensedSVG from '../assets/fonts/AIAEverest-Condensed.svg';

import AIACondensedMediumEOT from '../assets/fonts/AIAEverest-CondensedMedium.eot';
import AIACondensedMediumWOFF from '../assets/fonts/AIAEverest-CondensedMedium.woff';
import AIACondensedMediumWOFF2 from '../assets/fonts/AIAEverest-CondensedMedium.woff2';
import AIACondensedMediumTTF from '../assets/fonts/AIAEverest-CondensedMedium.ttf';
import AIACondensedMediumSVG from '../assets/fonts/AIAEverest-CondensedMedium.svg';

import AIAExtraBoldEOT from '../assets/fonts/AIAEverest-ExtraBold.eot';
import AIAExtraBoldWOFF from '../assets/fonts/AIAEverest-ExtraBold.woff';
import AIAExtraBoldWOFF2 from '../assets/fonts/AIAEverest-ExtraBold.woff2';
import AIAExtraBoldTTF from '../assets/fonts/AIAEverest-ExtraBold.ttf';
import AIAExtraBoldSVG from '../assets/fonts/AIAEverest-ExtraBold.svg';

import AIAMediumEOT from '../assets/fonts/AIAEverest-Medium.eot';
import AIAMediumWOFF from '../assets/fonts/AIAEverest-Medium.woff';
import AIAMediumWOFF2 from '../assets/fonts/AIAEverest-Medium.woff2';
import AIAMediumTTF from '../assets/fonts/AIAEverest-Medium.ttf';
import AIAMediumSVG from '../assets/fonts/AIAEverest-Medium.svg';

import AIARegularEOT from '../assets/fonts/AIAEverest-Regular.eot';
import AIARegularWOFF from '../assets/fonts/AIAEverest-Regular.woff';
import AIARegularWOFF2 from '../assets/fonts/AIAEverest-Regular.woff2';
import AIARegularTTF from '../assets/fonts/AIAEverest-Regular.ttf';
import AIARegularSVG from '../assets/fonts/AIAEverest-Regular.svg';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'AIABold';
    src: url('${AIABoldEOT}');
    src: url('${AIABoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIABoldWOFF}') format('woff'),
        url('${AIABoldWOFF2}') format('woff2'),
        url('${AIABoldTTF}') format('truetype'),
        url('${AIABoldSVG}#AIAEverestBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensed';
    src: url('${AIACondensedEOT}');
    src: url('${AIACondensedEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedWOFF}') format('woff'),
        url('${AIACondensedWOFF2}') format('woff2'),
        url('${AIACondensedTTF}') format('truetype'),
        url('${AIACondensedSVG}#AIAEverestCondensed') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensedMedium';
    src: url('${AIACondensedMediumEOT}');
    src: url('${AIACondensedMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedMediumWOFF}') format('woff'),
        url('${AIACondensedMediumWOFF2}') format('woff2'),
        url('${AIACondensedMediumTTF}') format('truetype'),
        url('${AIACondensedMediumSVG}#AIAEverestCondensedMedium') format('svg');
    font-weight: 500;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAExtraBold';
    src: url('${AIAExtraBoldEOT}');
    src: url('${AIAExtraBoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIAExtraBoldWOFF}') format('woff'),
        url('${AIAExtraBoldWOFF2}') format('woff2'),
        url('${AIAExtraBoldTTF}') format('truetype'),
        url('${AIAExtraBoldSVG}#AIAEverestExtraBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAMedium';
    src: url('${AIAMediumEOT}');
    src: url('${AIAMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIAMediumWOFF}') format('woff'),
        url('${AIAMediumWOFF2}') format('woff2'),
        url('${AIAMediumTTF}') format('truetype'),
        url('${AIAMediumSVG}#AIAEverestMedium') format('svg');
    font-weight: 600;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIARegular';
    src: url('${AIARegularEOT}');
    src: url('${AIARegularEOT}?#iefix') format('embedded-opentype'),
        url('${AIARegularWOFF}') format('woff'),
        url('${AIARegularWOFF2}') format('woff2'),
        url('${AIARegularTTF}') format('truetype'),
        url('${AIARegularSVG}#AIAEverestRegular') format('svg');
    font-weight: 400;
    font-style: normal;
  }

  body h1 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 96px;
    line-height: 84px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h2 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 72px;
    line-height: 64px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h3 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 48px;
    line-height: 44px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h4 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 32px;
    line-height: 28px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h5 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 24px;
    line-height: 22px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h6 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 20px;
    line-height: 20px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body p, ul, ol {
    font-family: "AIARegular", Arial, "Helvetica Neue", Helvetica, sans-serif;
    font-size: 15px;
    line-height: 20px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }
`;

export default GlobalStyle;
