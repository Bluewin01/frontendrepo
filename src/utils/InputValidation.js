export const validateIsInputEmpty = (props) => {
  const inputValues = Object.values(props);
  const inputKeys = Object.keys(props);
  let invalidInputList = [];

  inputValues.forEach((input, index) => {
    if (
      input === "" ||
      input === undefined ||
      input === null ||
      input.match(/^ *$/) !== null
    ) {
      invalidInputList.push(index);
    }
  });

  let validationResult = [];
  invalidInputList.forEach((fieldIndex) => {
    validationResult.push(inputKeys[fieldIndex]);
  });

  return validationResult;
};

export const validateIsInputListEmpty = (props) => {
  let invalidInputList = [];

  props.forEach((input) => {
    if (
      input.value === "" ||
      input.value === undefined ||
      input.value === null ||
      input.value.match(/^ *$/) !== null
    ) {
      invalidInputList.push(input.name);
    }
  });

  return invalidInputList;
};

export const validateEmailFormat = (props) => {
  const inputValues = Object.values(props);
  const inputKeys = Object.keys(props);
  let invalidEmailList = [];

  inputValues.forEach((email, index) => {
    if (
      !email.match(
        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
      )
    ) {
      invalidEmailList.push(index);
    }
  });

  let validationResult = [];
  invalidEmailList.forEach((fieldIndex) => {
    validationResult.push(inputKeys[fieldIndex]);
  });

  return validationResult;
};

export const validateEmailListFormat = (props) => {
  let invalidEmailList = [];

  props.forEach((email) => {
    if (email.value !== "") {
      if (
        !email.value.match(
          /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
        )
      ) {
        invalidEmailList.push(email.name);
      }
    }
  });

  return invalidEmailList;
};
