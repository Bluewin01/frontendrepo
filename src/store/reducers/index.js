import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
