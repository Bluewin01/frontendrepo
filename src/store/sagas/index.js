import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
