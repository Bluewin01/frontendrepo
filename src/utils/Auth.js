// Service to check authentication for user and to signOut
import * as AuthActions from "../store/Auth/AuthActions";
import store from "../store";

const Auth = {
  signOut() {
    store.dispatch(AuthActions.signOut);
  },
  isAuth(jwt) {
    if (!jwt) {
      return false;
    } else {
      return true;
    }
  },
};
export default Auth;
