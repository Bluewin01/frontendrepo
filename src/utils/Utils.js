// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "../store/reducers";

export function render(
  ui,
  { preloadedState, store = setupStore(preloadedState), ...renderOptions } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";
