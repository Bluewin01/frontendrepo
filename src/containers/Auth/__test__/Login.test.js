import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent, screen, act } from "../../../utils/Utils";
import renderer from "react-test-renderer";
import Login from "../Login";
import store from "../../../store";
import { Provider } from "react-redux";
import { waitForElement } from "@testing-library/react";

describe("Containers: Login", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it("should be render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Login />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Positive Case", () => {
  it("should can input agent userID and password", () => {
    render(<Login />);
    const mockUserID = "12345";
    const mockPassword = "abcd1234";
    const userID = screen.getByTestId("userID");
    const password = screen.getByTestId("password");
    const button = screen.queryByTestId("loginButton");
    act(() => {
      /* fire events that update state */
      fireEvent.change(userID, { target: { value: mockUserID } });
      fireEvent.change(password, { target: { value: mockPassword } });
      fireEvent.click(button);
    });
    expect(userID.value).toBe(mockUserID);
    expect(password.value).toBe(mockPassword);
    expect(button).toHaveTextContent("Login");
  });

  it("should can input staff userID and password", () => {
    render(<Login />);
    const mockUserID = "I012345";
    const mockPassword = "abcd1234";
    const userID = screen.getByTestId("userID");
    const password = screen.getByTestId("password");
    const button = screen.queryByTestId("loginButton");
    act(() => {
      /* fire events that update state */
      fireEvent.change(userID, { target: { value: mockUserID } });
      fireEvent.change(password, { target: { value: mockPassword } });
      fireEvent.click(button);
    });
    expect(userID.value).toBe(mockUserID);
    expect(password.value).toBe(mockPassword);
    expect(button).toHaveTextContent("Login");
  });
});

describe("Negative Case", () => {
  it("should not be able to input null for userID and password", async () => {
    render(<Login />);
    const mockUserID = "";
    const mockPassword = "";
    const userID = screen.getByTestId("userID");
    const password = screen.getByTestId("password");
    const button = screen.queryByTestId("loginButton");

    act(() => {
      /* fire events that update state */
      fireEvent.change(userID, { target: { value: mockUserID } });
      fireEvent.change(password, { target: { value: mockPassword } });
      fireEvent.click(button);
    });

    await waitForElement(() => screen.getByTestId("errorMessage"));
    expect(
      screen.getByText("Please fill in your ID or password")
    ).toBeInTheDocument();
  });

  it("should not be able to input userID less than 5 digits", async () => {
    render(<Login />);
    const mockUserID = "1234";
    const mockPassword = "mockPassword";
    const userID = screen.getByTestId("userID");
    const password = screen.getByTestId("password");
    const button = screen.queryByTestId("loginButton");

    act(() => {
      /* fire events that update state */
      fireEvent.change(userID, { target: { value: mockUserID } });
      fireEvent.change(password, { target: { value: mockPassword } });
      fireEvent.click(button);
    });

    await waitForElement(() => screen.getByTestId("errorMessage"));
    expect(screen.getByText("Invalid ID")).toBeInTheDocument();
  });

  it("should not be able to input userID with space", async () => {
    render(<Login />);
    const mockUserID = "123 45";
    const mockPassword = "mockPassword";
    const userID = screen.getByTestId("userID");
    const password = screen.getByTestId("password");
    const button = screen.queryByTestId("loginButton");

    act(() => {
      /* fire events that update state */
      fireEvent.change(userID, { target: { value: mockUserID } });
      fireEvent.change(password, { target: { value: mockPassword } });
      fireEvent.click(button);
    });

    await waitForElement(() => screen.getByTestId("errorMessage"));
    expect(screen.getByText("Invalid ID")).toBeInTheDocument();
  });
});
