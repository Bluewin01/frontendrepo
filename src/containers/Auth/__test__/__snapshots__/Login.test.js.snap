// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers: Login should be render correctly 1`] = `
<div
  className="login"
>
  <h3>
    Login to NTP 
  </h3>
  <form
    className="ant-form ant-form-vertical login__form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="input__form"
    >
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="userID"
            title="User ID"
          >
            User ID
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                autoComplete="off"
                className="ant-input"
                data-testid="userID"
                id="userID"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="User ID"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="input__form"
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="password"
            title="Password"
          >
            Password
          </label>
        </div>
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                autoComplete="off"
                className="ant-input"
                data-testid="password"
                id="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Password"
                type="password"
                value=""
                width="100%"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="input__form"
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <button
                className="ant-btn ant-btn-default ant-btn-lg ant-btn-dangerous btn__primary"
                data-testid="loginButton"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Login
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  
</div>
`;
